Do not expose and consider -logthreadnames when it does not work

hebasto opened the pull request:
There are conditions when the `HAVE_THREAD_LOCAL` macro is undefined what causes the `-logthreadnames` option does not work -- instead of thread names empty strings `[]` only are printed in the `debug.log` file.

This PR does not exposes the `-logthreadnames` option in such cases.

Refs:
- #16059
- #18652

hebasto committed:
Do not expose and consider -logthreadnames when it does not work

laanwj commented:
ACK 2ea5807fd36391e8f3b03c883066666132cfcb69

MarcoFalke commented:
Can you explain how the patch is supposed to work? Making an argument hidden in the help does not at all influence how it is parsed. The `[]` are still printed

hebasto commented:
Updated 2ea5807fd36391e8f3b03c883066666132cfcb69  -> b91e4ae0d8ab2ae6b77585c97c52d825f56ed539 ([pr18665.01](https://github.com/hebasto/bitcoin/commits/pr18665.01) -> [pr18665.02](https://github.com/hebasto/bitcoin/commits/pr18665.02), [diff](https://github.com/hebasto/bitcoin/compare/pr18665.01..pr18665.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/18665#issuecomment-616445550)

> Can you explain how the patch is supposed to work? Making an argument hidden in the help does not at all influence how it is parsed. The `[]` are still printed

MarcoFalke commented:
ACK b91e4ae0d8ab2ae6b77585c97c52d825f56ed539, looked at the diff, didn't test
